Provide fallback view sorting

cdinger opened the pull request:
This provides a default fallback sorting if a view is missing from the increasingly complex `view_dependencies` query. We've already encountered several edge cases were views are missing or misclassified in Oracle's `user_dependencies` view.

If a new edge case pops up this should not explode on a failed index lookup. Instead it will be added to the end of the sorted views array.

cdinger committed:
Provide fallback view sorting

This provides a default fallback sorting if a view is missing from the
increasingly complex `view_dependencies` query. We've already
encountered several edge cases were views are missing or misclassified
in Oracle's `user_dependencies` view.

If a new edge case pops up this should not explode on a failed index
lookup. Instead it will be added to the end of the sorted views array.

cdinger commented:
@whit0694 I'd like to get your thoughts on this. I don't like that the schema dumper fails catastrophically if a view is missing from the dependencies. This change ensures missing dependency views are added to the end of the dependency-sorted array.

Do you think these missing views should be added to the front or the back of the sorted array?

whit0694 commented:
Yeah, I can see that. 

I'm wondering if these changes will help stabilize the output of schema.rb. We get a lot of churn in that file, as the order of things seems to vary by environment. I'd like to make it more deterministic.

cdinger committed:
Deterministically sort views within dependencies

Views were being sorted base on their dependencies to other objects, but
within those contraints there was no specified order. This sorted views
alphabetically within the existing dependency ordering.

Any views that are missed in the dependency query are also sorted
alphabetically.

cdinger committed:
Memoize view_dependencies query

IanWhitney commented:
👍

cdinger committed:
Strip newlines and trailing semicolons from views

[Fixes #17]

cdinger commented:
@IanWhitney if I merge this into main would you be willing to test drive it on one of your projects?

IanWhitney commented:
Sure. Did you want to cut a release or have me grab the gem directly from the repo?

cdinger commented:
After I merge I'm hoping you can change your Gemfile to:

```
gem "scenic-oracle_adapter", git: "https://github.umn.edu/cdinger/scenic-oracle_adapter", branch: "main"
```

and give it a spin before we cut a new release.

IanWhitney commented:
Seems to be working as I'd expect. I'm able to build a schema.rb file and it looks like the order will be consistent. I think it's fine to release, though we may find things to change after more usage.
